cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -0,0 +1,197 @@
+# ~~~
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+# A function to add targets for GA libraries that use gRPC for transport.
+#
+# * library:      the short name of the library, e.g. `kms`.
+# * display_name: the display name of the library, e.g. "Cloud Key Management
+#   Service (KMS) API"
+# * GOOGLE_CLOUD_CPP_SERVICE_DIRS: a list of service directories within the
+#   library.
+#
+function (google_cloud_cpp_add_ga_grpc_library library display_name)
```

Comment:
This may be a case where a macro works better than a function.  Macros do not define a new scope.

Reply:
I don't think we need any variables defined in the inner scope to exist outside. So I think a `function` is appropriate.